File: 2022-declarative/domain/auth/src/main/kotlin/sample/domain/cache/model/PasswordResetRequestCache.java
package sample.domain.cache.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.TimeToLive;
import org.springframework.data.redis.core.index.Indexed;

public class PasswordResetRequestCache {
    @Id
    private String id;

    @Indexed
    private String code;

    @TimeToLive
    private Integer expire = 60 * 60 * 10;
}

File: 2021-rabbitmq-jpa/domain/redis/src/main/kotlin/sample/domain/cache/model/PasswordResetRequestCache.kt
package sample.domain.cache.model

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.TimeToLive
import org.springframework.data.redis.core.index.Indexed

class PasswordResetRequestCache(
    @Indexed
    val code: String,

    @TimeToLive
    val expire: Int = 60 * 60 * 10,
) {
    @Id
    var id: String? = null
}
